components/core/src/package/archive.rs

```diff
@@ -262,71 +248,53 @@ impl PackageArchive {
         self.read_deps(MetaFile::BuildTDeps)
     }
 
-    pub fn exposes(&mut self) -> Result<Vec<u16>> {
-        match self.read_metadata(MetaFile::Exposes) {
-            Ok(Some(data)) => {
-                let ports: Vec<u16> = data.split_whitespace()
-                                          .filter_map(|port| port.parse::<u16>().ok())
-                                          .collect();
-                Ok(ports)
-            }
-            Ok(None) => Ok(vec![]),
-            Err(e) => Err(e),
+    pub fn exposes(&mut self) -> Vec<u16> {
+        if let Some(data) = self.read_metadata(MetaFile::Exposes) {
+            data.split_whitespace()
+                .filter_map(|port| port.parse().ok())
```

Comment:
Even though the previous implementation did the same thing, it seems that it would be useful to at least log an error in case the metadata file existed but contained bogus contents that could not be parsed as a port.

Better still, perhaps this should still return a `Result`.

Reply:
Thanks, that is much better. Addressed in [521641](https://github.com/habitat-sh/habitat/pull/7710/commits/5216411dae740b5184c0e9efbb6aab96d8d50b67)